src/tsickle.ts

```diff
@@ -1461,18 +1461,33 @@ class Annotator extends ClosureRewriter {
    * Processes an EnumDeclaration into a Closure type. Always emits a Closure type, even in untyped
    * mode, as that should be harmless (it only ever uses the number type).
    */
-  private maybeProcessEnum(node: ts.EnumDeclaration) {
-    // Gather the members of enum, saving the constant value or
-    // initializer expression in the case of a non-constant value.
-    const members = new Map<string, number|ts.Node>();
-    let i = 0;
+  private processEnum(node: ts.EnumDeclaration) {
+    // Emit the enum declaration, which looks like:
+    //   /** @enum {number} */
+    //   const Foo = {BAR: 0, BAZ: 1, ...};
+    //   export {Foo};  // even if originally exported on one line.
+    // This declares an enum type for Closure Compiler (and Closure JS users of this TS code).
+    // Splitting the enum into declaration and export is required so that local references to the
+    // type resolve ("@type {Foo}").
+    this.emit('\n');
+    const name = node.name.getText();
+
+    this.emit(`/** @enum {number} */\n`);
+    this.emit(`const ${name}: any = {\n`);
+    // Emit enum values ('BAR: 0,').
+    let enumIndex = 0;
     for (const member of node.members) {
       const memberName = member.name.getText();
+      // Emit any comments and leading whitespace on the enum value definition.
+      this.writeRange(member, member.getFullStart(), member.getStart());
+      this.emit(`${memberName}: `);
+
       if (member.initializer) {
         const enumConstValue = this.typeChecker.getConstantValue(member);
         if (enumConstValue !== undefined) {
-          members.set(memberName, enumConstValue);
-          i = enumConstValue + 1;
+          this.emit(enumConstValue.toString());
+          // members.set(memberName, enumConstValue);
+          enumIndex = enumConstValue + 1;
```

Comment:
Since you're in here, I'll note that in TS 2.4 the enum const value can be a string.  I think there's nothing for you to to here yet but just FYI in case that guides your design any.